File: VB/WebSite/App_Code/XtraReport1.vb
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.Sql.DataApi
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing

''' <summary>
''' Summary description for XtraReport3
''' </summary>
Public Class XtraReport1
	Inherits DevExpress.XtraReports.UI.XtraReport

	Private TopMargin As TopMarginBand
	Private BottomMargin As BottomMarginBand
	Private Detail As DetailBand

	''' <summary>
	''' Required designer variable.
	''' </summary>
	Private components As System.ComponentModel.IContainer = Nothing

	Public Sub New()
		InitializeComponent()
		'
		' TODO: Add constructor logic here
		'
	End Sub
		Public Sub New(ByVal tableName As String)
		InitializeComponent()
		'
		' TODO: Add constructor logic here
		'

		Me.Bands.Add(New PageHeaderBand())

'INSTANT VB NOTE: The variable dataSource was renamed since Visual Basic does not handle local variables named the same as class members well:
		Dim dataSource_Renamed As New SqlDataSource("nwindConnectionString")
		Dim queryString As String = String.Format("SELECT * FROM {0}", tableName)
		Dim parameterQuery As SelectQuery = SelectQueryFluentBuilder.AddTable(tableName).SelectAllColumns().Build(tableName)
		dataSource_Renamed.Queries.Add(parameterQuery)
		dataSource_Renamed.RebuildResultSchema()
		dataSource_Renamed.Fill()
'INSTANT VB NOTE: The variable rowCount was renamed since Visual Basic does not handle local variables named the same as class members well:
		Dim rowCount_Renamed As Integer = DirectCast(dataSource_Renamed.Result(tableName), IList).Count

		Dim label As New XRLabel()
		label.Width = 500
		label.Font = New System.Drawing.Font("Verdana", 10F, FontStyle.Bold)
		Me.Bands(BandKind.PageHeader).Controls.Add(label)

		If rowCount_Renamed > 0 Then
'INSTANT VB NOTE: The variable padding was renamed since Visual Basic does not handle local variables named the same as class members well:
			Dim padding_Renamed As Integer = 10
			Dim tableWidth As Integer = Me.PageWidth - Me.Margins.Left - Me.Margins.Right - padding_Renamed * 2

			Dim dynamicTable As XRTable = XRTable.CreateTable(New Rectangle(padding_Renamed, 2, tableWidth, 40), 1, 0) ' table column count -  table row count -  height -  width -  rect Y -  rect X

			dynamicTable.Width = tableWidth
			dynamicTable.Rows.FirstRow.Width = tableWidth
			dynamicTable.Borders = DevExpress.XtraPrinting.BorderSide.All
			dynamicTable.BorderWidth = 1

			dynamicTable.BeginInit()
			For Each dc As IColumn In dataSource_Renamed.Result(tableName).Columns

				Dim cell As New XRTableCell()

				Dim binding As New ExpressionBinding("BeforePrint", "Text", dc.Name)
				cell.ExpressionBindings.Add(binding)
				cell.CanGrow = False
				cell.Width = 100
				cell.Text = dc.Name
				dynamicTable.Rows.FirstRow.Cells.Add(cell)
			Next dc
			dynamicTable.Font = New System.Drawing.Font("Verdana", 8F)
			dynamicTable.EndInit()
			Detail.Controls.Add(dynamicTable)

			label.Text = String.Format("Data table: {0}", tableName)

			Me.DataSource = dataSource_Renamed
			Me.DataMember = tableName
		Else
			label.Text = String.Format("There's no data to display or the table doesn't exist.")
		End If
		End Sub

	''' <summary> 
	''' Clean up any resources being used.
	''' </summary>
	''' <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
	Protected Overrides Sub Dispose(ByVal disposing As Boolean)
		If disposing AndAlso (components IsNot Nothing) Then
			components.Dispose()
		End If
		MyBase.Dispose(disposing)
	End Sub

	#Region "Designer generated code"

	''' <summary>
	''' Required method for Designer support - do not modify
	''' the contents of this method with the code editor.
	''' </summary>
	Private Sub InitializeComponent()
			Me.TopMargin = New DevExpress.XtraReports.UI.TopMarginBand()
			Me.BottomMargin = New DevExpress.XtraReports.UI.BottomMarginBand()
			Me.Detail = New DevExpress.XtraReports.UI.DetailBand()
			DirectCast(Me, System.ComponentModel.ISupportInitialize).BeginInit()
			' 
			' TopMargin
			' 
			Me.TopMargin.Name = "TopMargin"
			' 
			' BottomMargin
			' 
			Me.BottomMargin.Name = "BottomMargin"
			' 
			' Detail
			' 
			Me.Detail.Height = 0
			Me.Detail.Name = "Detail"
			' 
			' XtraReport3
			' 
			Me.Bands.AddRange(New DevExpress.XtraReports.UI.Band() { Me.TopMargin, Me.BottomMargin, Me.Detail})
			Me.PageHeight = 800
			Me.PageWidth = 1200
			Me.PaperKind = System.Drawing.Printing.PaperKind.Custom
		Me.Font = New System.Drawing.Font("Arial", 9.75F)
			Me.Version = "21.2"
			DirectCast(Me, System.ComponentModel.ISupportInitialize).EndInit()

	End Sub

	#End Region
End Class

File: CS/WebSite/App_Code/XtraReport1.cs
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.Sql.DataApi;
using DevExpress.XtraReports.UI;
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;

/// <summary>
/// Summary description for XtraReport3
/// </summary>
public class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
{
    private TopMarginBand TopMargin;
    private BottomMarginBand BottomMargin;
    private DetailBand Detail;

    /// <summary>
    /// Required designer variable.
    /// </summary>
    private System.ComponentModel.IContainer components = null;

    public XtraReport1()
    {
        InitializeComponent();
        //
        // TODO: Add constructor logic here
        //
    }
        public XtraReport1(string tableName)
    {
        InitializeComponent();
        //
        // TODO: Add constructor logic here
        //

        this.Bands.Add(new PageHeaderBand());

        SqlDataSource dataSource = new SqlDataSource("nwindConnectionString");
        string queryString = string.Format("SELECT * FROM {0}", tableName);
        SelectQuery parameterQuery = SelectQueryFluentBuilder
            .AddTable(tableName)
            .SelectAllColumns()
            .Build(tableName);
        dataSource.Queries.Add(parameterQuery);
        dataSource.RebuildResultSchema();
        dataSource.Fill();
        int rowCount = ((IList)dataSource.Result[tableName]).Count;

        XRLabel label = new XRLabel();
        label.Width = 500;
        label.Font = new System.Drawing.Font("Verdana", 10F, FontStyle.Bold);
        this.Bands[BandKind.PageHeader].Controls.Add(label);

        if (rowCount > 0)
        {
            int padding = 10;
            int tableWidth = this.PageWidth - this.Margins.Left - this.Margins.Right - padding * 2;

            XRTable dynamicTable = XRTable.CreateTable(
                                new Rectangle(padding,    // rect X
                                                2,          // rect Y
                                                tableWidth, // width
                                                40),        // height
                                                1,          // table row count
                                                0);         // table column count

            dynamicTable.Width = tableWidth;
            dynamicTable.Rows.FirstRow.Width = tableWidth;
            dynamicTable.Borders = DevExpress.XtraPrinting.BorderSide.All;
            dynamicTable.BorderWidth = 1;

            dynamicTable.BeginInit();
            foreach (IColumn dc in dataSource.Result[tableName].Columns)
            {

                XRTableCell cell = new XRTableCell();

                ExpressionBinding binding = new ExpressionBinding("BeforePrint", "Text", dc.Name);
                cell.ExpressionBindings.Add(binding);
                cell.CanGrow = false;
                cell.Width = 100;
                cell.Text = dc.Name;
                dynamicTable.Rows.FirstRow.Cells.Add(cell);
            }
            dynamicTable.Font = new System.Drawing.Font("Verdana", 8F);
            dynamicTable.EndInit();
            Detail.Controls.Add(dynamicTable);

            label.Text = string.Format("Data table: {0}", tableName);

            this.DataSource = dataSource;
            this.DataMember = tableName;
        }
        else
        {
            label.Text = string.Format("There's no data to display or the table doesn't exist.");
        }
    }

    /// <summary> 
    /// Clean up any resources being used.
    /// </summary>
    /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
    protected override void Dispose(bool disposing)
    {
        if (disposing && (components != null))
        {
            components.Dispose();
        }
        base.Dispose(disposing);
    }


    /// <summary>
    /// Required method for Designer support - do not modify
    /// the contents of this method with the code editor.
    /// </summary>
    private void InitializeComponent()
    {
            this.TopMargin = new DevExpress.XtraReports.UI.TopMarginBand();
            this.BottomMargin = new DevExpress.XtraReports.UI.BottomMarginBand();
            this.Detail = new DevExpress.XtraReports.UI.DetailBand();
            ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
            // 
            // TopMargin
            // 
            this.TopMargin.Name = "TopMargin";
            // 
            // BottomMargin
            // 
            this.BottomMargin.Name = "BottomMargin";
            // 
            // Detail
            // 
            this.Detail.Height = 0;
            this.Detail.Name = "Detail";
            // 
            // XtraReport3
            // 
            this.Bands.AddRange(new DevExpress.XtraReports.UI.Band[] {
            this.TopMargin,
            this.BottomMargin,
            this.Detail});
            this.PageHeight = 800;
            this.PageWidth = 1200;
            this.PaperKind = System.Drawing.Printing.PaperKind.Custom;
        this.Font = new System.Drawing.Font("Arial", 9.75F);
            this.Version = "21.2";
            ((System.ComponentModel.ISupportInitialize)(this)).EndInit();

    }

}
